### ISSUE TITLE
Perf: `chiERratprim` is a constant linear operator recomputed every iteration (1D path)

### ISSUE THREAD

**Issue by joglekara** [ergodicio/tsadar#118]
## Summary

`form_factor.py:280-282` recomputes a fixed linear operator on every forward pass:

```python
chiERratprim = vmap(ratintn.ratintn, in_axes=(None, 0, None))(
    ratdf, self.xi1[None, :] - self.xi2[:, None], self.xi1
)
```

`ratintn` is **exactly linear in its first argument `f`**, and at this call site the other two arguments (`g = xi1[None,:] - xi2[:,None]` and `z = xi1`) are built from `self.xi1` / `self.xi2`, which are fixed in `__init__`. So this block is a constant matrix applied to `ratdf`:

```
chiERratprim[:, 0] == M @ ratdf,   M fixed, shape (1640, 1024)
```

Only `ratdf` varies across fit iterations. `M` can be built once.

## Evidence

Linearity of `ratintn` in `f`, verified numerically:

```
f(3a + 2.5b) vs 3f(a) + 2.5f(b):  max rel err 1.7e-12
f(0):                             0.0 exactly
```

This follows from reading `ratintn.py`: `fdif`, `fav` are linear stencils in `f`; `tmp = fav*gdif - gav*fdif` is linear; `rf` and `rfn` are linear; `jnp.where` and `jnp.real` preserve linearity over the reals. `gdif`, `gav`, `use_rf` depend only on `g`.

Measured (JAX CPU, x64, `npts` grid as in `__init__`: `xi1` 1024, `xi2` 1640):

| | current | `M @ ratdf` | speedup |
|---|---|---|---|
| forward | 26.7 ms | 0.31 ms | 87x |
| gradient | 69.0 ms | 0.61 ms | 113x |

Agreement: **1.5e-14 absolute**, on output values of order 4. This is exact to roundoff, not an approximation.

## Suggested implementation

Build `M` once in `FormFactor.__init__`. Don't use `jacfwd` (it works, but took 25 s) — there's a closed form. `ratcen` can be written as

```
out = P * f_av + Q * f_dif
```

with `P`, `Q` depending only on `g`:

```
rf  = fav*(1/gav_s + gdif**2/(12*gav_s**3)) + fdif*(-gav*gdif/(12*gav_s**3))
rfn = fav*(L/gdif)                          + fdif*(1/gdif - gav*L/gdif**2)
```

where `L = log((gav + 0.5j*gdif)/(gav - 0.5*gdif))`. Since `f_av` and `f_dif` are two-point stencils, `M` assembles with two scatter-adds in a single O(1640 x 1024) pass.

## Notes

- Worth a regression test pinning `M @ ratdf` against the current implementation at a few `ratdf` values.
- The 2D/ARTS path calls `ratintn` with `g = vx - xie_mag_at`, which varies per evaluation point — `M` is **not** shared with that path. See the separate `custom_vjp` issue for that side.
- Benchmarked on CPU (no GPU available in the session). The win here is removing ~1.7M complex `log`/division evaluations per forward pass, so it should hold or improve on GPU.
